The robo kitten polo patch

TheRoboKitten opened the pull request:
Needs reviewing and clean-up

Commit:
Poloniex Patch Part 1

Poloniex Patch Part 1

Commit:
Updated engine.js with some things removed

Updated engine.js with some things removed (Sorry for bad desc... very tired.)

Commit:
Updated poloniex.js 0.0.8

You might need to turn off strict ssl but beware in poloniex.js. /zenbot/node_modules/poloniex.js/lib/poloniex.js

Commit:
Files for hosting back-testing reports

Make sure to use apache2. lighttpd wont work. not sure about httpd

Commit:
Updated conf

The conf I'm using, basically same thing.

Commit:
Coin Trading script that switches coin

Script that will switch between profitable pairs of coins based on a comparison of price at two points. (very very basic...)

Commit:
Delete exchange.js

Commit:
Updated exchange.js

crubb reviewed (changes requested):
As discussed in #developers/Discord, please separate the backtester and polo stuff.

Commit:
Update conf-sample.js

Commit:
Delete algo.py

Commit:
Delete index.php

Commit:
Delete index.php.graph

Commit:
Delete start.sh

Commit:
Delete trend_ema.py

Commit:
Delete cointrader.py

Commit:
Update README.md

Commit:
Update README.md

DeviaVir commented on README.md:
```diff
@@ -370,7 +370,8 @@ Attempts to buy low and sell high by tracking RSI high-water readings.
 - Risky to use in bear markets, since the algorithm depends on price recovery.
-- If the other strategies are losing you money, this strategy may perform better, since it basically "reverses the signals" and anticipates a reversal instead of expecting the trend to continue.
+- If the other strategies are losing you money, this strategy may perform better, since it basically "reverses the signals" and
+anticipates a reversal instead of expecting the trend to continue.
```
why a newline?

DeviaVir commented on README.md:
```diff
@@ -460,6 +461,17 @@ zenbot sell gdax.BTC-USD --pct=10
+  - Removed buy re-buy on_hold in engine.js (this seemed to fix something.)
+  - Removed Order On Hold wait forbuy function. (temporary!) (would be great to just set a timeout wait for buy syncbalance do_reoder that works)
+  - Apparently fixed some calculations with order on hold.
+  - This needs debugging and testing.
```
I'm not sure what this is doing here?

DeviaVir commented on conf-sample.js:
```diff
@@ -25,8 +26,8 @@ c.gdax.passphrase = 'YOUR-PASSPHRASE'
-c.poloniex.key = 'YOUR-API-KEY'
-c.poloniex.secret = 'YOUR-SECRET'
+c.poloniex.key = ''
+c.poloniex.secret = ''
```
why?

DeviaVir commented on conf-sample.js:
```diff
@@ -85,10 +86,10 @@ c.profit_stop_pct = 1
-c.sell_pct = 99
-// ms to adjust non-filled order after
+c.sell_pct = 100
+// ms to adjust non-filled order after (This might need debugging)
```
there's a reason for this

DeviaVir commented on conf-sample.js:
```diff
@@ -99,12 +100,12 @@ c.wait_for_settlement = 5000
 c.markup_pct = 0
 // become a market taker (high fees) or a market maker (low fees)
-c.order_type = 'maker'
+c.order_type = 'taker'
```
why?

DeviaVir commented on conf-sample.js:
```diff
@@ -99,12 +100,12 @@ c.wait_for_settlement = 5000
 
 // default # days for backfill and sim commands
-c.days = 14
+c.days = 30
```
why?

DeviaVir commented on lib/engine.js:
```diff
@@ -251,17 +251,17 @@ module.exports = function container (get, set, clear) {
+                   // }
+                   //else {
+                   cb(null, do_reorder ? null : false)
+                   // }
```
why are you commenting this?

DeviaVir commented on lib/engine.js:
```diff
@@ -457,7 +457,7 @@ module.exports = function container (get, set, clear) {
                   }
                 }
-                if (n(s.balance.asset).subtract(s.balance.asset_hold || 0).value() < n(size).value()) {
+                if (n(s.balance.asset || 0).value() < n(s.balance.asset_hold).value()) {
```
what do these changes do for you?

DeviaVir commented on package.json:
```diff
@@ -27,7 +27,7 @@
     "number-abbreviate": "^2.0.0",
     "numbro": "git+https://github.com/carlos8f/numbro.git",
-    "poloniex.js": "0.0.7",
+    "poloniex.js": "git+https://github.com/premasagar/poloniex.js.git",
```
this is a horrible idea, at least point it to a commit.

Commit:
Update engine.js

Commit:
Merge pull request #2 from TheRoboKitten/TheRoboKitten-polo-patch

The robo kitten polo patch

Commit:
Update engine.js

Commit:
Merge pull request #3 from TheRoboKitten/Engine.Js-Syncbalance-timeout

Update engine.js

Commit:
Set a timeout for syncbalance. 

Yes... this number is very very high.. 10 seconds. What can you do, their API is too slow.

Commit:
Increased order adjust....

Increasing order adjust to keep buy/sell orders open longer to compensate for slow poloniex order api sync.

Commit:
Update conf-sample.js

Commit:
hard-coded and increased timeouts even more..

Seriously how laggy is poloniex?

5s is not enough time to update their order API!

I hard coded this to 10 seconds. This might get us somewhere. But understandable, this can be lowered for other exchanges. I dont expec this pull to be merged at this point, I just have it up so you guys can pull ideas from.

Talk to you guys later, life calls.

Commit:
Merge branch 'master' into TheRoboKitten-polo-patch

txssseal commented:
Also might be a good idea to clean up the commits @TheRoboKitten

Commit:
Pointed poloniex.js to a commit.

Added a specific commit to poloniex.js (tags are outdated on his master, releases are too old)

Commit:
Removed old notes, will use comments

Will use github comments for code notes.

Commit:
Fixed newline

Fixed the newline

Commit:
Reset conf.js

I reccomend a line describing to set your buy percent to 10% and sell percent to 100% and using RSI strategy if running a bot for every coin on poloniex.

ghost commented:
Please disregard everything in this PR. I don't want to merge this, its too unstable and dated.

Plus, a lot of these things could probably compare to the speed branch which I would like to focus on more :)

Sorry.
